app/views/layouts/comfy/admin/cms/_nav_main.html.haml

```diff
@@ -18,8 +18,20 @@
 
           %li.nav-primary__item
           %li.nav-primary__item.nav-primary__item--right
+            = link_to "#{content_tag(:span, '', class: 'nav-primary__icon fa fa-search')}".html_safe, '#', class: 'nav-primary__text nav-primary__text--button', data: { dough_component: 'Collapsable', dough_collapsable_trigger: 'search' }
+          %li.nav-primary__item.nav-primary__item--right
             = link_to "#{current_user.name} #{content_tag(:span, '', class: 'nav-primary__icon fa fa-user')}".html_safe, edit_user_path(current_user), class: 'nav-primary__text'
           %li.nav-primary__item.nav-primary__item--right
             = link_to "#{t('users.sign_out')} #{content_tag(:span, '', class: 'nav-primary__icon fa fa-sign-out')}".html_safe, destroy_user_session_path, method: :delete, class: 'nav-primary__text'
 
-        = cms_hook :navigation
+    - if @site.persisted? && @site.present?
```

Comment:
This will fail if @site is nil, because nil doesn't have persisted? method. This needs to be inverted.